tests/unit_tests/db/test_db_handler.py

```diff
@@ -76,360 +163,239 @@ def test_find_latest_simulation_model_db(db, db_no_config_file, mocker):
+        db, "_read_mongo_db", return_value={"parameter": "value"}
     )
-    logger.info("Now test inserting the same file again, this time expect a warning")
-    with caplog.at_level(logging.WARNING):
-        file_id = db.insert_file_to_db(file_name, f"sandbox_{random_id}")
-    assert "exists in the DB. Returning its ID" in caplog.text
-    assert (
-        file_id == db._get_file_mongo_db(f"sandbox_{random_id}", f"test_file_{random_id}.dat")._id
+
+    parameter = "mirror_list"
+    parameter_version = "1.0.0"
+    site = None
+    array_element_name = "LSTS-01"
+    collection = "telescopes"
+
+    result = db.get_model_parameter(
+        parameter, parameter_version, site, array_element_name, collection
     )
 
+    query = {
+        "parameter_version": parameter_version,
+        "parameter": parameter,
+        "instrument": array_element_name,
+    }
 
-def test_get_all_versions(db, mocker, caplog):
+    mock_read_mongo_db.assert_called_once_with(query=query, collection_name=collection)
+    assert result == {"parameter": "value"}
 
-    # not specifying any database names, collections, or parameters
-    all_versions = db.get_all_versions()
-    assert all(_v in all_versions for _v in ["5.0.0", "6.0.0"])
-    assert any(key.endswith("None") for key in db.model_versions_cached)
 
-    # not specifying a telescope model name and parameter
-    all_versions = db.get_all_versions(
-        array_element_name=None,
-        site="North",
-        parameter=None,
-        collection="telescopes",
+def test_get_model_parameter_no_array_element_name(db, mocker):
+    """Test get_model_parameter method without array element name."""
+    mock_read_mongo_db = mocker.patch.object(
+        db, "_read_mongo_db", return_value={"parameter": "value"}
     )
-    assert all(_v in all_versions for _v in ["5.0.0", "6.0.0"])
-    assert any("telescopes" in key for key in db.model_versions_cached)
 
-    # using a specific parameter
-    all_versions = db.get_all_versions(
-        array_element_name="LSTN-01",
-        site="North",
-        parameter="camera_config_file",
-        collection="telescopes",
+    parameter = "corsika_iact_io_buffer"
+    parameter_version = "10.11.12"
+    site = "South"
+    array_element_name = None
+    collection = "configuration_corsika"
+
+    result = db.get_model_parameter(
+        parameter, parameter_version, site, array_element_name, collection
+    )
+
+    query = {
+        "parameter_version": parameter_version,
+        "parameter": parameter,
+        "site": site,
+    }
+
+    mock_read_mongo_db.assert_called_once_with(query=query, collection_name=collection)
+    assert result == {"parameter": "value"}
+
+
+def test_get_model_parameters(db, mocker):
+    """Test get_model_parameters method."""
+    mock_get_production_table = mocker.patch.object(
+        db,
+        "_read_production_table_from_mongo_db",
+        return_value={"parameters": {"LSTN-01": {"param1": "v1"}}},
```

Comment:
Realistic values?

Edit: There are too many examples below, impossible to come up with realistic values for all, so you can forget it.

Reply:
I know what you mean, not ideal. But would change the tests, so dummy values should be fine.